articles/mysql-query-performance-optimization-tips/index.md

```diff
@@ -0,0 +1,180 @@
+- Avoid Null Values. Declare columns to be NOT NULL where possible. This enables better use of indexes. NULL values increase the processing power needed for testing whether each value is NULL, making SQL operations slower.
+
+#### Use Indexes
+Think of records as content in a book. If you want to learn on a particular subtopic, you would go to the index pages, look for the subtopic you want, then get the page where the subtopic is. Indexes work the same way. They are used to find rows with specific column values much faster. Without using an index, MySQL must begin with searching in the first row and go through the whole table to find the required records. Tables with a huge amount of data are more costly to query.
+
+With the use of an index, MySQL can faster determine the position to seek in the middle of the data file. This is done without going through all the rows and is much faster than reading every row sequentially. Refer to [MySQL developer](https://dev.mysql.com/doc/refman/8.0/en/mysql-indexes.html) guide for more information on indexes.
+
+You can create a single-column or multiple column indexes, as shown below, respectively.
+```sql
+CREATE INDEX tablename_columnname_idx ON tablename (columnname);
+```
+```sql
+CREATE INDEX tablename_column1name_column2name_idx ON tablename (column1name, column2name);
+```
+
+#### Use Wildcards at the End of a Phrase
+In MySQL, wildcards are used in conjunction with the LIKE operator and NOT LIKE operator. They are used to search for data matching some search criteria. You can learn more about wildcards [here](https://www.guru99.com/wildcards.html).
+
+Wildcards result in the most expansive scan when searching for data, which is very inefficient. Leading wildcards are the most inefficient, especially when combined with ending wild cards. In such a case, MySQL has to search all the records for a match. Thus you should avoid leading wild cards. See the queries bellow, one using a leading wildcard and another one using ending wildcards.
+
+```sql
+SELECT * FROM city WHERE name LIKE '%Al%';
+```
+
+```sql
+SELECT * FROM city WHERE name LIKE 'Al%';
+```
+
+#### Specify Columns in SELECT Function
+SELECT * (select all) is used as a shortcut to query all columns available in a table.  This requires more resources than using a SELECT statement with only the columns you need for that specific query. For example, a customer table with 20 different columns and a hundred thousand entries. If you want to select a city with ID and Name only; use
+
+```sql
+SELECT ID, Name, District FROM city;
+```
+
+instead of
+
+```sql
+SELECT * FROM city;
+```
+
+The second example will take more time to run to completion.
```

Comment:
I like this format where you say: 
use <SQL> instead of <SQL>. 
The reason why. 

I would suggest using this in the Use Wildcards at the End of a Phrase, Avoid SELECT DISTINCT, and Use LIMIT sections. My only other suggestion would be to lead with "try to use" instead of "use" because sometimes it is just not possible for one reason or another.